### ISSUE TITLE
Refactor e2e tests

### ISSUE THREAD

**Issue by kaglowka** [PrzypisPowszechny/pp-client#365]
- [ ] fix urls so it is DRY
- [ ] try to avoid restarting browser after most of the tests (and reset the browser state by clearing existing tabs etc.)
- other?

**Comment by glowka** [comment]
> try to avoid restarting browser after most of the tests (and reset the browser state by clearing existing tabs etc.)

This is jest characteristic of jest that one spec file = one process. So browser is killed after a spec is finished.
The common solution is to run related tests in one spec file and do not split the cases between files to easily.

**Comment by kaglowka** [comment]
> > try to avoid restarting browser after most of the tests (and reset the browser state by clearing existing tabs etc.)
> 
> This is jest characteristic of jest that one spec file = one process. By default jest run in multiple processes (in hour case this is limited by server port) and one spec file is a thread unit here, so the browser being killed when a spec is finished is really deep-rooted.
> The common solution is to run related tests in one spec file and do not split the cases between files to easily.

No, sorry, I didn't think there was another meaning to this point. I meant something else.

Currently in multi-tab e2e tests (i.e. including popup) I've made browser restart in **beforeEach** rather than **beforeAll**. This slows these test case considerably, of course.

**Comment by glowka** [comment]
Oh, Ok
